audit-service/src/unit-tests/java/uk/gov/gchq/palisade/service/audit/ArgumentCaptorAssert.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018-2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.audit;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.ListAssert;
+import org.mockito.ArgumentCaptor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Assertion methods for {@code ResponseEntity}s.
+ */
+public class ArgumentCaptorAssert<T> extends AbstractAssert<ArgumentCaptorAssert<T>, ArgumentCaptor<T>> {
```

Comment:
why have you made your own assertion captor? AssertJ has plenty of these, why create our own?

Reply:
Makes tests easier to read. Using vanilla AssertJ you cannot do `assertThat(captor).isEmpty()` as the assertJ would return an ObjectAssert instance. You could also not do `assertThat(captor).containsOnly()`. I'm hoping this will be the basis of another 1:30pm (short) discussion. I only did a couple of custom assertions. There is scope for a lot more. If used wisely can reduce quite a lot of test code.